pages/verification/index.tsx

```diff
@@ -0,0 +1,169 @@
+            toast('humanID verification failed, please try again', {
+                autoClose: 3000,
+                type: 'error',
+            })
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    return (
+        <BaseContainer className="bg-black">
+            <Helmet>
+                <title>Human ID Verification</title>
+            </Helmet>
+            <LayoutContainer>
+                <div className="h-full flex flex-col justify-between">
+                    <LoaderWrapper isLoading={isLoading}>
+                        <div className="flex-1-0-0 relative overflow-hidden rounded-2xl">
+                            <img className="h-full w-full absolute object-cover" alt="verification image" src="/image/Verification_Illustration.svg" />
+                        </div>
+                        <div className="pt-4 gap-y-4 flex flex-col h-min">
+                            <text className="font-inter font-medium text-2xl text-justify">To send this message, please verify that you’re not a bot.</text>
+                            <div className="border border-gray06 rounded-lg rounded-t-xl flex flex-col">
+                                <button className="shadow-3xl bg-humanId_blue" style={{
+                                    margin: '-1px',
+                                    backgroundImage: `url('/image/humanID_bg_button.svg')`,
+                                    backgroundSize: 'contain',
+                                    backgroundRepeat: 'no-repeat',
+                                    backgroundPosition: 'center',
+                                    height: '45px',
+                                    borderRadius: '8px'
+                                }} onClick={() => getLoginPage()} />
+                                <div className="flex flex-col m-3">
+                                    <text className="text-center font-inter font-semibold">What is humanID?</text>
+                                    <text className="text-center font-normal text-gray06 text-base">Created by the <span className="text-foundationBlue" onClick={() => router.push(publicRuntimeConfig.HUMAN_INTERNET_URL)}>Foundation for a Human Internet</span>, humanID verifies that you’re not a bot without storing your data or sharing it with BetterSocial</text>
+                                </div>
+                            </div>
+                        </div>
+                    </LoaderWrapper>
+                </div>
+            </LayoutContainer>
+        </BaseContainer>
+    );
+}
+
+export const getServerSideProps = (context: GetServerSidePropsContext) => {
+    const query = context.query; // Retrieve the URL parameter from context.query
+
+    return {
+        props: {
+            exchangeToken: query['sent_message?et'] || '',
+            isSendMessage: !!query['sent_message?et'] || false,
+            isFailedVerify: query['login_failed'] === '' || false,
+        },
+    };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function `getServerSideProps` does not handle potential errors when accessing properties of `context.query`. If `context.query` is undefined, it will throw an error. It's recommended to add checks before accessing its properties.
```diff
export const getServerSideProps = (context: GetServerSidePropsContext) => {
    const query = context.query; // Retrieve the URL parameter from context.query

    return {
        props: {
            exchangeToken: query['sent_message?et'] || '',
            isSendMessage: !!query['sent_message?et'] || false,
            isFailedVerify: query['login_failed'] === '' || false,
        },
    };
};
+    if (!context || !context.query) {
+        // Handle the error here
+    }
```

Please note that these are suggestions for improvement and the current implementation may still work as expected.


<!-- This is an auto-generated comment by OSS CodeRabbit -->